client/src/app/pages/applications/application-assessment/components/application-assessment-wizard/application-assessment-wizard.tsx

```diff
@@ -184,76 +189,191 @@ export const ApplicationAssessmentWizard: React.FC<
-          // const commentValue = commentValues[fieldName];
-          return {
-            ...section,
-            // comment: commentValue,
-            questions: section.questions.map((question) => ({
+    const updatedQuestionsData = formValues[QUESTIONS_KEY];
+
+    // Create an array of sections based on the questionsData
+    const sections: Section[] =
+      matchingQuestionnaire?.sections?.map((section) => {
+        // const commentValues = values["comments"];
+        // const fieldName = getCommentFieldName(category, false);
+        // const commentValue = commentValues[fieldName];
+        return {
+          ...section,
+          // comment: commentValue,
+          questions: section.questions.map((question) => {
+            return {
               ...question,
               answers: question.answers.map((option) => {
-                const questionValues = values["questions"];
-
                 const fieldName = getQuestionFieldName(question, false);
-                const questionValue = questionValues[fieldName];
+                const questionAnswerValue = updatedQuestionsData[fieldName];
                 return {
                   ...option,
-                  selected: questionValue === option.text,
+                  selected: questionAnswerValue === option.text,
                 };
               }),
-            })),
-          };
-        }) || [],
-      status: assessmentStatus,
-    };
-
-    patchAssessment(payload)
-      .then(() => {
-        switch (saveAction) {
-          case SAVE_ACTION_VALUE.SAVE:
-            history.push(Paths.applications);
-            break;
-          case SAVE_ACTION_VALUE.SAVE_AND_REVIEW:
-            assessment?.application?.id &&
-              getApplicationById(assessment.application.id)
-                .then((data) => {
-                  history.push(
-                    formatPath(Paths.applicationsReview, {
-                      applicationId: data.id,
-                    })
-                  );
-                })
-                .catch((error) => {
-                  pushNotification({
-                    title: getAxiosErrorMessage(error),
-                    variant: "danger",
-                  });
-                });
-            break;
-        }
-      })
-      .catch((error) => {
-        console.log("Save assessment error:", error);
+            };
+          }),
+        };
+      }) || [];
+    return sections;
+  };
+
+  const handleSaveAsDraft = async (
+    formValues: ApplicationAssessmentWizardValues
+  ) => {
+    try {
+      if (!assessment?.application?.id) {
+        console.log("An assessment must exist in order to save as draft");
+        return;
+      }
+      const sections = assessment
+        ? buildSectionsFromFormValues(formValues)
+        : [];
+
+      const payload: Assessment = {
+        ...assessment,
+        sections,
+        status: "STARTED",
+      };
+
+      await patchAssessment(payload);
```

Comment:
I wonder if we want to switch to using a `useMutation` here eventually for these  `patchAssessment` calls so we can separate API logic from the component. Sounds out of scope for this PR though since it's a preexisting pattern.

Reply:
Updated.